cmd/execution_builder.go

```diff
@@ -277,8 +277,8 @@ func (exeNode *ExecutionNode) LoadGCPBlockDataUploader(
 		exeNode.events = storage.NewEvents(node.Metrics.Cache, node.DB)
 		exeNode.commits = storage.NewCommits(node.Metrics.Cache, node.DB)
 		exeNode.computationResultUploadStatus = storage.NewComputationResultUploadStatus(node.DB)
-		exeNode.collections = storage.NewCollections(node.DB, exeNode.transactions)
 		exeNode.transactions = storage.NewTransactions(node.Metrics.Cache, node.DB)
+		exeNode.collections = storage.NewCollections(node.DB, exeNode.transactions)
```

Comment:
somehow my local testings (localnet) didn't catch this failure..

Reply:
some of these should already be available via `exeNode.Storage.*` 
https://github.com/onflow/flow-go/blob/master/cmd/scaffold.go#L723

Reply:
confirmed with @m4ksio , in this context we are using the storage instances from `ExecutionNode` itself instead from scaffold. Also in `execution_builder.go` the FlowNodeBuilder.Build() is not called as in other types of node builders (access\observer\follower_builder.go) in which the storage instances in scaffold are initialized.

Reply:
`FlowNodeBuilder.Build()` is called [here](https://github.com/onflow/flow-go/blob/master/cmd/execution/main.go#L18-L18)

Are we using new versions intentionally here? I think using existing instances is best when possible as it uses less resources and avoids gotcha's in instantiation like we hit here.

Reply:
yes that's what i meant. `ExecutionNodeBuilder` has all module\components initialization task in `LoadComponentsAndModules()` instead of a `Build()` as other node builders where they call `FlowNodeBuilder.Build()`. 

However to get all `exeNode.Storage.*` ready when needed like at this line here, `FlowNodeBuilder.Build()` needs to be called before node-specific initialization(in our case `LoadComponentsAndModules()`). Sth can be refactored more in this part but I'd say we can do it in a separate PR.

Reply:
synced offline with @peterargue. so `exeNode.Storage.*` should have been ready in `LoadComponentsAndModules()` since those `Module()` and `Component()` are actually callback. But we do need to revisit why `ExecutionNodeBuilder` does not reuse what's already available in `FlowNodeBuilder`. Some clean up can be done here but probably in a separate PR.